### PR TITLE
Add va-notification web component

### PR DESCRIPTION
**Pull request by allisonlu**
## Chromatic

https://va-notification--60f9b557105290003b387cd5.chromatic.com

---
## Configuring this pull request
- [x] Link to any related issues in the description so they can be cross-referenced.
- [x] Under **Testing done**, be specific about how this update was tested. 
    - Examples: Storybook, Chrome, Browserstack, Mobile/Responsive, etc.
- [x] Add the appropriate version patch label (`major`, `minor`, `patch`, or `ignore-for-release`). 
    - See [How to choose a version number](https://github.com/department-of-veterans-affairs/component-library#how-to-choose-a-version-number) for guidance.
    - Use `ignore-for-release` if files haven't been changed in a component library package. (ie. only Storybook files)
- [ ] DST Only: Increment the `/packages/core` version number if this will be the last PR merged before a release.
- [x] Complete all sections below.
- [x] Delete this section once complete

## Description
Structure of the `va-notification` web component and its story is based on those of `va-alert`. 

Closes [issue #57714](https://github.com/department-of-veterans-affairs/va.gov-team/issues/57714) on the My VA board.

Other relevant links:
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1181
- Sketch file (Desktop): [single](https://www.sketch.com/s/9b0e6efc-423a-4354-9db3-ab2083d566c9/a/ZOkzKM7), [multiple](https://www.sketch.com/s/9b0e6efc-423a-4354-9db3-ab2083d566c9/a/zxDMl7W)
- Sketch file (Mobile): [single](https://www.sketch.com/s/9b0e6efc-423a-4354-9db3-ab2083d566c9/a/YG34bgd), [multiple](https://www.sketch.com/s/9b0e6efc-423a-4354-9db3-ab2083d566c9/a/JnjwPe5)

## Testing done
- locally, visually on Storybook
- e2e tests pass

## Screenshots

Screenshot 2023-05-16 at 3 00 24 PM

Screenshot 2023-05-16 at 3 00 38 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Comment by jamigibbs** [comment]
> Should I implement analytics tracking in this PR, or wait until this approved first?

That's a great questions! I think we might want to create a separate analytics event for this component (suppressing the others).

But I believe that would be out of scope at the moment. I wonder can/should we make a separate ticket for adding analytics to this new component, @caw310? I'm not sure what the procedure was in the past but I feel like we had done that as a separate piece of work after the component was merged.

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-notification/va-notification.tsx:110]: I think we might want this to be a link instead of a button. Do the notifications links usually navigate to another page? https://design.va.gov/components/button/#links-vs-buttons

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-notification/va-notification.tsx:117]: Is there a design reference for this close text? 
Screenshot 2023-05-23 at 1 53 17 PM

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-notification/va-notification.tsx:98]: You can probably just leave out the `="true"` part. I think just the presence of the prop implies truthiness.

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-notification/va-notification.tsx:35]: Will there ever be other icons besides action-required and update? If someone passes another icon value into this prop, would it display any Font Awesome icon that matches that class name?

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (allisonlu) [reply packages/web-components/src/components/va-notification/va-notification.tsx:117]: yes! @humancompanion-usds and @aagosto90 discussed building variations with the Close text: 

https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1181#issuecomment-1458632802

File: packages/storybook/stories/va-notification.stories.jsx
Comment (jamigibbs) [inline packages/storybook/stories/va-notification.stories.jsx:98]: Does the notification component always have a dateTime display like this? 

I'm just wondering if this should either be made its own prop or we maybe just make a dedicated named slot for it that we style from within the component if possible (we probably don't want Formation classes used because we're trying to move off of that framework).

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-notification/va-notification.tsx:15]: Could probably be:

```
 * @maturityCategory caution
 * @maturityLevel candidate
```

File: packages/storybook/stories/va-notification.stories.jsx
Comment (jamigibbs) [reply packages/storybook/stories/va-notification.stories.jsx:93]: I had another thought about managing the header. I'm assuming (someone from design can confirm) that we want the header text to be styled the same regardless of the header level (ie. font-size). So that might be all the reason we need to bring this in as either a named slot that we style in the component to display a consistent font size for all header levels, or, create two separate props for managing the header; one for the text and one for the header size. 

Examples for either of those approaches:

- Using props for the header in [va-radio](https://design.va.gov/storybook/?path=/docs/components-va-radio--label-header)
- Using slot for the header in [va-accordion](https://design.va.gov/storybook/?path=/docs/components-va-accordion--change-header-level)

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (jamigibbs) [reply packages/web-components/src/components/va-notification/va-notification.tsx:117]: @humancompanion-usds @aagosto90 @babsdenney The only Sketch file we have for QA'ing is this one I think: https://www.sketch.com/s/9b0e6efc-423a-4354-9db3-ab2083d566c9/a/ygvmgOq#Inspect

Do we happen to have any other source of truth or are we ok with using that one for now? There is no variation for the "close" text currently.

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (jamigibbs) [inline packages/web-components/src/components/va-notification/va-notification.tsx:45]: Do we want to make this button functional for the actual visibility of the component? I'm thinking similar to how we have a `visible` prop in [va-banner](https://design.va.gov/storybook/?path=/docs/components-va-banner--dismissible) and [va-modal](https://design.va.gov/storybook/?path=/docs/components-va-modal--default).

**Comment by jamigibbs** [comment]
We have a separate ticket created for adding google analytics here: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1803

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (Andrew565) [inline packages/web-components/src/components/va-notification/va-notification.tsx:93]: If `headlineLevel` is a non-numeric string, this would result in `null` being returned, which would prevent a headline from displaying. Is this the desired effect? Would it be better to return a `h3` by default and have L#119 below show or not-show based on the value of the `headline` prop?

File: packages/storybook/stories/va-notification.stories.jsx
Comment (jamigibbs) [reply packages/storybook/stories/va-notification.stories.jsx:92]: @allisonlu We had some internal back and forth about this and what we'd like to do is create a named slot for the date/time. I've mocked something up just to make sure I wasn't leading you down the wrong path. But essentially what we'd like to do is manage the styling of this element in the component itself while also allowing some slot flexibility with content itself.

Here's an approach which I think does that but do you mind giving it a try? We essentially want to remove the need for the `p` or any styling classes outside of the component.

Screenshot 2023-06-01 at 2 46 44 PM

Screenshot 2023-06-01 at 2 46 59 PM

Screenshot 2023-06-01 at 2 46 48 PM

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (ajakabcin) [reply packages/web-components/src/components/va-notification/va-notification.tsx:135]: @jamigibbs here you go - sorry about that these files includes the text and the x icon: [desktop](https://www.sketch.com/s/9b0e6efc-423a-4354-9db3-ab2083d566c9/a/ZOkzKM7), [mobile](https://www.sketch.com/s/9b0e6efc-423a-4354-9db3-ab2083d566c9/a/YG34bgd)

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (ajakabcin) [reply packages/web-components/src/components/va-notification/va-notification.tsx:35]: we don't anticipate at this time other types of icons, but I defer to @allisonlu as to whether it would be possible to do so if someone wanted to

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (ajakabcin) [reply packages/web-components/src/components/va-notification/va-notification.tsx:135]: @jamigibbs I updated the sketch file links in the body above as well

**Review by Andrew565 — CHANGES_REQUESTED** [review]
Fix required for headline.

**Review by jamigibbs — APPROVED** [review]
All of my feedback has been addressed. Thank you @allisonlu!

File: packages/web-components/src/components/va-notification/va-notification.tsx
Comment (Andrew565) [inline packages/web-components/src/components/va-notification/va-notification.tsx:119]: This needs to be `{headline ?` not `{HeadlineLevel`, since HeadlineLevel will always resolve to 'something' but `headline` is not guaranteed to be non-null.

**Review by Andrew565 — CHANGES_REQUESTED** [review]

**Review by babsdenney — APPROVED** [review]
The design looks great! It looks like it uses all the correct colors and font styles.  Some things to keep in mind for future changes.
- Moving the [close button above the title](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1181#issuecomment-1447073790) 
- [Improving the interaction of the close button](https://www.sketch.com/s/be292806-8dbf-4201-bfa7-0edbba103347/a/4aYE093)

**Review by Andrew565 — APPROVED** [review]

**Pull request merged**
